Arduino 101 dts

SavinayDharmappa opened the pull request.

jhedberg commented on dts/x86/arduino_101.dts:
```diff
@@ -0,0 +1,29 @@
+	status = "ok";
+	current-speed = <115200>;
+};
+
```
The current arduino_101_defconfig has the following:

 CONFIG_UART_QMSI_0_BAUDRATE=1000000
 CONFIG_UART_QMSI_0_HW_FC=y

Shouldn't arduino_101.dts default to those as well for uart0? Where else will the right values get set? UART0 is the UART between the Quark SE and the nRF51, so it's critical for BLE functionality that the configuration is correct.

SavinayDharmappa replied:
yes it should. To add hw-flowcontrol uart.yaml in has to be changed in dts/common/. I would add uart0 support in arduino.dts in next patch. These set of patch intention was to set the baudrate for uart1 from device tree instead of Kconfig.

jhedberg commented on dts/x86/yaml/arduino101.yaml:
```diff
@@ -0,0 +1,42 @@
+version: 0.1
+
+description: >
+    This binding gives a base representation of the K64 UART
```
K64?

SavinayDharmappa replied:
corrected

nashif commented on dts/x86/arduino_101.dts:
```diff
@@ -0,0 +1,29 @@
+#include "intel_curie.dtsi"
+
+/ {
+	model = "Intel Curie Arduino 101 board";
```
just call this Arduino 101, Arduino 101 is not an Intel product.

SavinayDharmappa replied:
okay.

nashif commented on dts/x86/arduino_101.fixup:
```diff
@@ -0,0 +1,5 @@
+
+#define CONFIG_UART_QMSI_1_BAUDRATE	INTEL_ARDUNIO101_UART_B0002400_CURRENT_SPEED
```
ARDUINO_101_, skip INTEL_

SavinayDharmappa replied:
okay.

nashif reviewed (approved):
+

nashif reviewed (dismissed):
+

agross-linaro commented on dts/x86/intel_curie.dtsi:
```diff
@@ -0,0 +1,59 @@
+
+		uart0: uart@B0002000 {
+			compatible = "intel,ardunio101-uart";
+			reg = <0xB0002000 0x400>;
```
A nit: addresses use lower case letters.  So 0xB0002000 should be 0xb0002000.

SavinayDharmappa replied:
incorporated the changes

agross-linaro commented on dts/x86/intel_curie.dtsi:
```diff
@@ -0,0 +1,59 @@
+		};
+
+		uart0: uart@B0002000 {
+			compatible = "intel,ardunio101-uart";
```
mispelling here: intel,arduino101-uart.

SavinayDharmappa replied:
corrected

agross-linaro commented on dts/x86/yaml/arduino101.yaml:
```diff
@@ -0,0 +1,42 @@
+---
+title: ardunio101 Uart
+id: intel,ardunio-uart
```
Arduino?

SavinayDharmappa replied:
corrected

agross-linaro commented on dts/x86/yaml/arduino101.yaml:
```diff
@@ -0,0 +1,42 @@
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "intel,ardunio101-uart"
```
Same thing here.  I assume this was supposed to be  arduino.  At least these were all consistent

SavinayDharmappa replied:
corrected

jhedberg commented on dts/x86/arduino_101.dts:
```diff
@@ -0,0 +1,29 @@
+
+&uart0 {
+	status = "ok";
+	current-speed = <1000000>;
```
This looks better now. What about the HW flow control that was previously enabled through Kconfig? Is that still coming from there?

galak replied:
I believe HW flow control would come from Kconfig, if we want to do this via DT, we should look to see if the 'uart-has-rtscts' property as defined by Linux Documentation/devicetree/bindings/serial/serial.txt makes sense

SavinayDharmappa replied:
yes HW flowcontrol is comming from Kconfig. DTS support of  HW flowcontrol requires changes in  dts/common/yaml/uart.yaml so i will push those changes as new patch.

tbursztyka commented on dts/x86/mem.h:
```diff
@@ -0,0 +1,9 @@
+
+#define DT_FLASH_SIZE		__SIZE_K(144)
+#define DT_SRAM_SIZE		__SIZE_K(55)
+
```
Isn't these valid for arduino only? (I am asking that because the file seems generic to all x86)

SavinayDharmappa replied:
It is specific to Quark SE C1000.

galak commented on arch/x86/core/Kconfig:
```diff
@@ -53,12 +53,14 @@ config MAX_IRQ_LINES
 config PHYS_LOAD_ADDR
 	hex "Physical load address"
 	# Default value must be supplied by platform
+	#depends on !HAS_DTS
```
I think the # before depends is wrong

galak commented on dts/x86/arduino_101.dts:
```diff
@@ -0,0 +1,29 @@
+
+/ {
+	model = "Arduino 101";
+	compatible = "intel,quark";
```
should be: compatible = "arduino,101", "intel,quark";

SavinayDharmappa replied:
okay.

galak commented on dts/x86/intel_curie.dtsi:
```diff
@@ -0,0 +1,59 @@
+		};
+
+		cpu@1 {
+			compatible = "arch";
```
What are you trying to do with compatible = arch?

SavinayDharmappa replied:
It should be arc. I will correct it.

galak commented on dts/x86/intel_curie.dtsi:
```diff
@@ -0,0 +1,59 @@
+		};
+	};
+
+	flash0: flash {
```
since at fixed address:

flash0: flash@40010000 {

SavinayDharmappa replied:
okay

galak commented on dts/x86/intel_curie.dtsi:
```diff
@@ -0,0 +1,59 @@
+	};
+
+
+	sram0: memory {
```
since at a fixed location:

sram0: memory@a8006400 {

SavinayDharmappa replied:
okay

galak commented on dts/x86/yaml/arduino101.yaml:
```diff
@@ -0,0 +1,42 @@
+---
+title: arduino101 Uart
```
Is the uart really arduino101 specific, isn't the UART QMSI?  This should probably be "Intel QMSI Uart" (fixup other uart references accordingly)

galak commented on drivers/serial/Kconfig.qmsi:
```diff
@@ -51,7 +51,7 @@ config UART_QMSI_1_NAME
 
 config UART_QMSI_1_BAUDRATE
-	depends on UART_QMSI_1
+	depends on !HAS_DTS
```
You should get the 'name' and 'baudrate' for all QMSI UARTs from the DT.

SavinayDharmappa replied:
okay

galak commented on dts/x86/arduino_101.fixup:
```diff
@@ -1,7 +1,5 @@
-#define CONFIG_UART_QMSI_0_BAUDRATE	INTEL_ARDUNIO101_UART_B0002000_CURRENT_SPEED
```
Why are these removed in this commit?

SavinayDharmappa replied:
without sanity check i pushed for first time. It caused warning in build so removed and pushed it agin

galak replied:
ok, little weird, lets get the other things in order than we can figure this out.

galak reviewed (changes requested):
A number of cleanups / changes needed.

galak commented on dts/x86/yaml/arduino101.yaml:
```diff
@@ -0,0 +1,42 @@
+---
```
Add this in the patch to updates the serial driver.

galak commented on dts/x86/yaml/arduino101.yaml:
```diff
@@ -0,0 +1,42 @@
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "arduino_101-uart"
```
Why is this arduino_101-uart, and not intel,qmsi-uart?

galak commented on dts/x86/intel_curie.dtsi:
```diff
@@ -0,0 +1,61 @@
+		};
+
+		uart1: uart@b0002400 {
+			compatible = "arduino_101-uart";
```
compatiable = "intel,qmsi-uart"?

galak commented on dts/x86/intel_curie.dtsi:
```diff
@@ -0,0 +1,61 @@
+		};
+
+		uart0: uart@b0002000 {
+			compatible = "arduino_101-uart";
```
compatiable = "intel,qmsi-uart"?

galak commented on drivers/serial/Kconfig.qmsi:
```diff
@@ -43,15 +44,15 @@ config UART_QMSI_1
 
 config UART_QMSI_1_NAME
-	depends on UART_QMSI_1
+	depends on !HAS_DTS
```
Leave the depends on as is, lets do the same if !HAS_DTS

galak reviewed (changes requested):
Some changes for UART, lets split out the flow control patch into a separate PR.

galak reviewed (changes requested):
In general things are looking good, however I would restructure the patches so we have:
1. changes to serial driver
2. changes to x86 in general
3. changes that add arduino 101 support
(anything to boards/x86/arduino_101 or has arduino_101 in the name).

SavinayDharmappa committed:
dts: x86: Add dts support for x86

patch adds necessary files and does the modification to the existing
files to add device support for x86 based intel quark microcontroller

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

SavinayDharmappa committed:
drivers: serial: Add device support for serial driver

patch enables qmsi serial driver to pick baudrate and name
from device tree instead of Kconfig for UART0 and UART1
instance

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>

SavinayDharmappa committed:
dts: arduino101: Add device tree support for arduino101 board

patch adds necessary files and does the modification
to the existing files to add device support for
arduino_101 board.

Signed-off-by: Savinay Dharmappa <savinay.dharmappa@intel.com>
